Kconfig.zephyr

```diff
@@ -378,6 +378,14 @@ config COMPILER_SAVE_TEMPS
 	  Instruct the compiler to save the temporary intermediate files
 	  permanently. These can be useful for troubleshooting build issues.
 
+config COMPILER_NO_TRACK_MACRO_EXPANSION
+	bool "Do not track macro expansion"
+	help
+	  Do not track locations of tokens across macro expansions. While the
+	  compiler diagnostics will be reduced, it can be useful in situations
+	  where the expansion chain is long enough to _hide_ useful error
+	  messages.
```

Comment:
It is arguably more intuitive to make this `COMPILER_TRACK_MACRO_EXPANSION` and `default y`.

Reply:
I vote to change as well

Reply:
done!